R/mobr.R

```diff
@@ -427,7 +427,305 @@ samp_ssad = function(comm, groups){
+                  following tests will be performed:', tests_string))
+    }
+    return(approved_tests)
+}
+
+# Auxillary function for get_delta_stats()
+# Perform checks when type is "discrete"
+get_delta_discrete_checks = function(ref_group, group_levels, group_data, env_var){
+    if (is.null(ref_group)) {
+        stop('For a discrete analysis you must specify a ref_group to compare groups to')
+    } else if (!(as.character(ref_group) %in% group_levels)) {
+        stop('Reference group does not exist.')
+    }
+    if (!is.null(env_var))
+        warning('Environmental variable is not used in the discrete analysis.')
+    if (!('factor' %in% class(group_data))) 
+        warning('Grouping variable is not a factor. A group will be defined for
+                each unique value.')
+}
+
+# Auxillary function for get_delta_stats()
+# Perform checks when type is "continuous"
+get_delta_continuous_checks = function(corr, group_levels, env_levels){
+    if (!(corr %in% c('spearman', 'pearson')))
+        stop('corr has to be spearman or pearson.')
+    if ('factor' %in% class(env_levels)) {
+        env_vals = data.frame(groups = group_levels, 
+                              values = as.numeric(env_levels)[match(group_levels, env_levels)])
+        warning('Variable of interest is a factor but will be treated as a continous 
+                variable for the analysis with the above values')
+        print(env_vals)
+    } 
+}
+
+# Auxillary function for get_delta_stats()
+# Returns a vector of abundances where individual-based rarefaction 
+# will be performed
+get_delta_ind_sample = function(group_sad, inds, log_scale){
+    group_minN = min(rowSums(group_sad))
+    if (is.null(inds)){
+        ind_sample_size = seq(group_minN)
+    } else if (length(inds) > 1) {
+        ind_sample_size = inds
+        if (max(inds) > group_minN)
+            warning('Sample size is higher than abundance of at least one group!')
+    } else if (log_scale == T){
+        ind_sample_size = floor(exp(seq(inds) * log(group_minN) / inds))
+    } else {
+        ind_sample_size = floor(seq(inds) * group_minN / inds)
+    }
+    ind_sample_size = unique(c(1, ind_sample_size)) # Force (1, 1) to be included
+    return(ind_sample_size)
+}
+
+# Auxillary function for get_delta_stats()
+# Returns the "assumed" plot density given 
+# whether min, max or mean is used
+get_plot_dens = function(site_sp_mat, density_stat){
+    if (density_stat == 'mean') {
+        plot_dens = sum(site_sp_mat) / nrow(site_sp_mat)
+    } else if (density_stat == 'max') {
+        plot_dens = max(rowSums(site_sp_mat))
+    } else {
+        plot_dens = min(rowSums(site_sp_mat))
+    }
+    return(plot_dens)   
+}
+
+# Auxillary function for get_delta_stats()
+# Obtain the swap curve and/or spatial curve for each group if asked
+# Directly add attributes to the input "out"
+get_sample_curves = function(out, comm, group_levels, approved_tests){
```

Comment:
Unless there is a good reason that I can't see you should not list `out` as one of the arguments. It does not appear that the function relies on any information from this argument. This will help to simplify your code because you will just have to define an object called `out` that is a null data.frame at first.